src/app/loaders/express/error-handler.ts

```diff
@@ -22,9 +24,41 @@ const isHTTPLikeError = (
   return types.isNativeError(err) && 'statusCode' in err
 }
 
-// 404 since no middleware responded
-export const catchNonExistentRoutesMiddleware: RequestHandler = (_req, res) => {
-  res.sendStatus(StatusCodes.NOT_FOUND)
+// If static no middleware responded
+export const catchNonExistentStaticRoutesMiddleware: RequestHandler = async (
+  req,
+  res,
+) => {
+  // Attempt to fetch from s3 bucket
+
+  try {
+    const { data } = await axios.get(
+      `${config.aws.staticAssetsBucketUrl}${req.originalUrl}`,
+      {
+        responseType: 'stream',
+      },
+    )
+    // If get request succeeds pipe the data back
+    logger.info({
+      message: 'Serving static asset from S3',
+      meta: {
+        action: 'catchNonExistentStaticRoutesMiddleware',
+        url: req.originalUrl,
+      },
+    })
+    data.pipe(res)
+  } catch (err) {
+    // Else return 404
+    logger.error({
```

Comment:
The error here is not necessarily a 404, correct? which is why we are logging?

404s will be logged by express anyways and we'll know it comes from here.

Reply:
Yup - in fact it will be a 403 from s3 if the item doesn't exist in s3

Reply:
Added doc in `[012ce00](https://github.com/opengovsg/FormSG/pull/5595/commits/012ce00898f4398cccccf9abca2752572015e0cb)`